File: veloProWeb/src/main/java/com/veloProWeb/Validation/SupplierValidator.java
package com.veloProWeb.Validation;

import com.veloProWeb.Model.Entity.Purchase.Supplier;
import org.springframework.stereotype.Component;

@Component
public class SupplierValidator {
    public void validate(Supplier supplier){
        validateName(supplier.getName());
        validatePhone(supplier.getPhone());
        validateEmail(supplier.getEmail());
        validateRut(supplier.getRut());
    }

    private void validateName(String name){
        if (name == null || name.trim().isBlank() || name.trim().length() < 3) {
            throw new IllegalArgumentException("Ingrese un nombre válido.");
        }
    }

    private void validatePhone(String phone){
        if (phone == null || phone.trim().isBlank() || phone.trim().length() != 13) {
            throw new IllegalArgumentException("Ingrese número válido, Ej: +569 12345678");
        }
    }

    private void validateEmail(String email){
        if (email.trim().isBlank() || !email.matches("[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}")) {
            throw new IllegalArgumentException("Ingrese Email válido.");
        }
    }

    private void validateRut(String rut){
        if (rut.trim().isBlank() || !rut.matches("^\\d{7,8}-\\d|[kK]$")) {
            throw new IllegalArgumentException("Rut no es correcto, Ingrese un formato válido");
        }
    }
}

File: veloProweb-frontend/src/app/validation/supplier-validator.ts
import { SupplierForm } from "../models/Entity/Purchase/supplier-form";

export class SupplierValidator {
    static validateForm(selectedSupplier: SupplierForm): boolean{
        return this.isFieldValid(selectedSupplier, 'name') &&
            this.isFieldValid(selectedSupplier, 'rut') &&
            this.isFieldValid(selectedSupplier, 'phone') &&
            this.isFieldValid(selectedSupplier, 'email');
    }

    static isFieldValid(supplier: SupplierForm, fieldName: keyof SupplierForm): boolean{
        const value = supplier[fieldName];
        if (fieldName === 'name') {
            return value !== null && typeof value === 'string' && value.trim().length > 2;
        }
        if (fieldName === 'rut') {
            return value !== null && typeof value === 'string' && /^\d{7,8}-\d|[kK]$$/.test(value);
        }
        if (fieldName === 'phone') {
            return value !== null && typeof value === 'string' && value.trim().length == 13;
        }
        if (fieldName === 'email') {
            return !value || typeof value === 'string' && /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(value);
        }
        return true;
    }
}
